util.c

```diff
@@ -227,6 +227,28 @@ check_firmware_version(YK_KEY *yk, bool verbose, bool quiet)
 }
 
 int
+check_user_group(char *username, char *group)
+{
+  struct group *group_ptr;
+  char **group_member;
+  
+  if((group_ptr = getgrnam(group)) == NULL) {
```

Comment:
Please use getgrnam_r() instead of getgrnam() here. One issue here is that group membership defined in the user record is not checked by the getgrnam functions, that has to be fetched with getpwnam_r() and getgrgid_r().